homeassistant/components/samsungtv/config_flow.py

```diff
@@ -154,6 +155,8 @@ async def _async_get_and_check_device_info(self):
         self._udn = _strip_uuid(dev_info.get("udn", info["id"]))
         if mac := mac_from_device_info(info):
             self._mac = mac
+        elif mac := getmac.get_mac_address(ip=self._host):
+            self._mac = mac
```

Comment:
This line in the config flow needs tests

Reply:
Indeed, will add as soon as I have a bit of time for it ;-)